runtime/include/mem/jemalloc/chpl-mem-impl.h

```diff
@@ -29,6 +29,11 @@
 
 #define MALLOCX_NO_FLAGS 0
 
+
+extern bool chpl_mem_impl_allocLocalizes;
+#define CHPL_MEM_IMPL_ALLOCLOCALIZES chpl_mem_impl_allocLocalizes
```

Comment:
This is really confusing to me. Can we just have a chpl_mem_impl_allocLocalizes() routines that each mem layer must implement. And for jemalloc have it be `strcmp(CHPL_COMM, "ugni") != 0 || (num_heaps = chpl_topo_getNumNumaDomains()) <= 1` and then reuse that code in chpl_mem_layerInit?